openfisca_core/entities.py

```diff
@@ -51,9 +51,21 @@ def check_role_validity(self, role):
         if role is not None and not type(role) == Role:
             raise Exception("{} is not a valid role".format(role))
 
-    def __call__(self, variable_name, period, options = [], **parameters):
+    def __call__(self, variable_name, period=None, options = [], **parameters):
         self.check_variable_defined_for_entity(variable_name)
 
+        if period is None:
+            import traceback
```

Comment:
We usually import everything on the top of the module, except when you really can't do it (circular import).

If you're worried about performances, see [these](http://stackoverflow.com/questions/1188640/good-or-bad-practice-in-python-import-in-the-middle-of-a-file) [discussions](http://stackoverflow.com/questions/128478/should-python-import-statements-always-be-at-the-top-of-a-module).